### PR TITLE
feat: major rework of Helm Chart

### PR DESCRIPTION
**Pull request by duhow**
requires Kubernetes >= 1.18 (which is default nowadays, latest available is 1.31)

- fix "app.fullname" to avoid duplication of name installation (eg. part-db-part-db)
- get app.url from Ingress or default Service name
- move ENV vars to Secret
- change Deployment to StatefulSet
- use StatefulSet volume claim template instead of creating PVC
- allow disable persistence (empty folder)
- change Ingress to default apiVersion v1

- add initContainer to perform database migration / initialization
- add initContainer to log the password (termination log, done only once when first starting the application)
- enable livenessProbe
- mount persistent database folder
- remove non-standard default storageClassName
- reduce default persistence size to 4GB (can be enlarged)

In short, this Chart can be installed and used right away without requiring user interaction (besides getting the first admin password)

❗ Note this is breaking change, hence bumping `minor` version.

### CODE REVIEW

**Comment by jbtronics** [comment]
Thank you very much. 
I do not have much experience with helm charts (and the initial version of the chart were not written by me), but this looks and works fine and is a useful rework.

If you want to write some documentation on how to use this chart, feel free to create a PR for it. As you might have seen, the current documentation is really rudimentary, and it is also not advertised in the Part-DB main documentation.
